File: languages/php/Bloock/Entity/Identity/Method.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents an enumeration of methods used in the DID.
 */
class Method
{
    const IDEN3 = "IDEN3";
    const POLYGON_ID = "POLYGON_ID";

    public static function fromProto(\Bloock\Method $method): string
    {
        switch ($method) {
            case \Bloock\Method::IDEN3:
                return Method::IDEN3;
            case \Bloock\Method::POLYGON_ID:
                return Method::POLYGON_ID;
            default:
                return Method::POLYGON_ID;
        }
    }

    public static function toProto(string $method): int
    {
        switch ($method) {
            case Method::IDEN3:
                return \Bloock\Method::IDEN3;
            case Method::POLYGON_ID:
                return \Bloock\Method::POLYGON_ID;
            default:
                return \Bloock\Method::POLYGON_ID;
        }
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity/Method.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.IdentityEntities;

/**
 * Represents an enumeration of methods used in the DID.
 */
public enum Method {
  IDEN3,
  POLYGONID;

  public IdentityEntities.Method toProto() {
    switch (this) {
      case IDEN3:
        return IdentityEntities.Method.IDEN3;
      case POLYGONID:
        return IdentityEntities.Method.POLYGON_ID;
      default:
        return IdentityEntities.Method.POLYGON_ID;
    }
  }
}
